underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -197,6 +197,11 @@ public Object getConf() {
   @Override
   public List<String> getFileLocations(String path, FileLocationOptions options)
       throws IOException {
+    // If the user has hinted the underlying storage nodes are not co-located with Alluxio
+    // workers, short circuit without querying the locations
+    if (Configuration.getBoolean(PropertyKey.UNDERFS_HDFS_REMOTE)) {
+      return null;
```

Comment:
We should probably specify the extraordinary return values in the interface javadoc